ZShare/ViewModels/ExtensionViewModel.swift

```diff
@@ -737,42 +737,69 @@ final class ExtensionViewModel {
         DDLogInfo("ExtensionViewModel: downloaded unsupported attachment")
 
         // Remove downloaded file, it won't be used anymore
-        try? self.fileStorage.remove(file)
+        try? fileStorage.remove(file)
 
         guard (url.host ?? "").contains("sciencedirect") else {
-            self.state.attachmentState = .failed(.downloadedFileNotPdf)
+            state.attachmentState = .failed(.downloadedFileNotPdf)
             return
         }
 
         // Try loading the url in webview to bypass redirects
 
         DDLogInfo("ExtensionViewModel: detected sciencedirect, trying redirect")
 
-        self.state.attachmentState = .downloading(0)
-        self.state.retryCount += 1
-
-        self.getRedirectedPdfUrl(from: url) { [weak self] newUrl, newCookies, newUserAgent, newReferrer in
-            guard let self = self else { return }
+        state.attachmentState = .downloading(0)
+        state.retryCount += 1
 
-            if let newUrl = newUrl, newUrl != url && self.state.retryCount < 3 {
-                self.download(item: item, attachment: attachment, attachmentUrl: newUrl, to: file, cookies: newCookies, userAgent: newUserAgent, referrer: newReferrer)
+        getRedirectedPdfUrl(from: url, timeout: .seconds(2)) { [weak self] newUrl, newCookies, newUserAgent, newReferrer in
+            guard let self else { return }
+            if let newUrl, newUrl != url && state.retryCount < 3 {
```

Comment:
As far as I tested, the html file already downloaded (the one removed before), does not contain the final page. So we need to follow the redirects first, in order to check for a potential captcha form.

The completion handler is called with non-nil values when a PDF is actually extracted. Otherwise, when tuple values are nil, we then check if there is a captcha form. If there is one, we load the URL again (with allowed redirects), with the web view visible, and a larger timeout, to allow for the user to complete the captcha.

I'll simplify further, by not stopping the load of the web view in `RedirectWebViewHandler` on timeout, so we can just make the already loaded web view visible, by just adding another handler for the timeout case.